src/module/system/check/check.ts

```diff
@@ -627,13 +666,26 @@ class CheckPF2e {
      * @param roll  The roll that is to be rerendered
      * @param isOld This is the old roll render, so remove damage or other buttons
      */
-    static async renderReroll(roll: Rolled<Roll>, { isOld }: { isOld: boolean }): Promise<string> {
+    static async renderReroll(
+        roll: Rolled<Roll>,
+        { isOld, resource }: { isOld: boolean; resource?: RerollOptions["resource"] },
+    ): Promise<string> {
         const die = roll.dice.find((d): d is Die => d instanceof foundry.dice.terms.Die && d.faces === 20);
         if (typeof die?.total !== "number") throw ErrorPF2e("Unexpected error inspecting d20 term");
 
         const html = await roll.render();
         const element = parseHTML(`<div>${html}</div>`);
 
+        // Handle roll formula visibility
+        const formulaElement = htmlQuery(element, ".dice-formula");
+        if (formulaElement && !isOld && resource !== "mythic-points") {
+            // Hide rerolled formula for everything except mythic point rerolls
+            formulaElement.classList.add("hidden");
```

Comment:
I tried using `formulaElement.hidden = true` but that was stripped somewhere along the ways. The CSS class seems to work without problems.